api/deployments/models/component_builder.go

```diff
@@ -128,8 +126,19 @@ func (b *componentBuilder) WithComponent(component v1.RadixCommonDeployComponent
 		}
 	}
 
-	if auth := component.GetAuthentication(); auth != nil && component.GetPublicPort() != "" && deployment.IsSecretRequiredForClientCertificate(auth.ClientCertificate) {
-		b.secrets = append(b.secrets, utils.GetComponentClientCertificateSecretName(component.GetName()))
+	if auth := component.GetAuthentication(); auth != nil && component.IsPublic() {
+		if deployment.IsSecretRequiredForClientCertificate(auth.ClientCertificate) {
```

Comment:
Agree. It has been like this since client cert authentication was introduced.
The props in certificateConfig returned by parseClientCertificateConfiguration in IsSecretRequiredForClientCertificate is never nil, but this "we" don't know that, and should therefore do a nil check. I have added this as a bug to the backlog.
`certificateConfig := parseClientCertificateConfiguration(*clientCertificate)`